.github/workflows/NUnitConsoleAndEngine.CI.yml

```diff
@@ -1,118 +1,84 @@
 ﻿name: NUnitConsoleAndEngine.CI
 
 on:
-  push:
-    branches:
-      - main
-      - release
-      - version3x
   pull_request:
-  
+    branches-ignore:
+      - "azure-*"
+    paths-ignore:
+      - "*.txt"
+      - "*.md"
+
 env:
   DOTNET_NOLOGO: true                     # Disable the .NET logo
   DOTNET_CLI_TELEMETRY_OPTOUT: true       # Disable sending .NET CLI telemetry  
 
 jobs:
-  build-windows:
-    name: Windows Build
-    runs-on: windows-latest
+  build:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        include:
+          - os: windows-latest
+            testRunName: Windows
+          - os: ubuntu-latest
+            testRunName: Linux
+          - os: macos-14
+            testRunName: Linux
+      fail-fast: false
```

Comment:
I'm thinking we may want one or more jobs to run after all the builds under the matrix. 

The matrix builds could do all compilation, testing and package creation and testing.

If all (or possibly selected) builds succeeded, a publish job could actually publish the results.

If all that passed, a Release job could perform a production release.

Reply:
Yeah, I think there are a few options to do that.  Either have a portion of the job that only runs on selected branches to publish after merges, or have something that triggers a separate workflow to do the publish.

But either way, unless we go whole hog into Actions or make some major adjustments to the cake recipe it's not as nice because we won't be able to reuse the results from the build: e.g. if we run `dotnet cake --target=Package` and check if it succeeds and then run `dotnet cake --target=Publish` [or whatever it actually is], then cake will re-run the build/test/package steps again.  If everything was in Actions then we would run the package stuff, save the build artifacts, and then separately run the publish step which would just reference them.